src/components/StoreRevokeBlock.js

```diff
@@ -1,80 +1,162 @@
+import { getIsVerifying } from "../reducers/application";
+import {
+  getRevokeCertificateHash,
+  updateRevokeCertificateHash
+} from "../reducers/admin";
+
+const certificateDropzone = css`
+  width: 100%;
+  text-align: center;
+`;
 
 class StoreRevokeBlock extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      certificateHash: "",
-      certificateHashIsValid: true
+      certificate: {},
+      certificateHashIsValid: true,
+      isModalVisible: false
     };
-
     this.onHashChange = this.onHashChange.bind(this);
     this.onRevokeClick = this.onRevokeClick.bind(this);
+    this.handleCertificateSelected = this.handleCertificateSelected.bind(this);
+    this.toggleModal = this.toggleModal.bind(this);
+  }
+
+  getCertificateStatus(certValidator) {
+    const validaor = this.props.revokeCertificateValidity[certValidator];
+    const status = {
+      verified: validaor ? validaor.valid : true,
+      verifying: false,
+      error: ""
+    };
+    return status;
   }
 
   onHashChange(event) {
+    const inputCertificateHash = event.target.value;
+    this.props.updateRevokeCertificateHash(inputCertificateHash);
+  }
+
+  toggleModal() {
+    const { isModalVisible } = this.state;
+    const { revokeCertificateHash } = this.props;
     this.setState({
-      certificateHash: event.target.value,
-      certificateHashIsValid: isValidCertificateHash(event.target.value)
+      certificateHashIsValid: isValidCertificateHash(revokeCertificateHash)
     });
+    if (isValidCertificateHash(revokeCertificateHash)) {
+      this.setState({
+        isModalVisible: !isModalVisible
+      });
+    }
+  }
+
+  handleCertificateSelected(payload) {
+    this.props.verifyCertificateValidity(payload);
   }
 
   onRevokeClick() {
-    const { adminAddress, storeAddress, handleCertificateRevoke } = this.props;
-    const { certificateHash } = this.state;
+    const {
+      adminAddress,
+      storeAddress,
+      handleCertificateRevoke,
+      revokeCertificateHash
+    } = this.props;
+    if (isValidCertificateHash(revokeCertificateHash)) {
+      const payload = {
+        storeAddress,
+        fromAddress: adminAddress,
+        certificateHash: revokeCertificateHash
+      };
+      handleCertificateRevoke(payload);
+      this.toggleModal();
+    }
+  }
 
-    this.setState({
-      certificateHashIsValid: isValidCertificateHash(certificateHash)
-    });
-    if (isValidCertificateHash(certificateHash)) {
-      // eslint-disable-next-line no-alert
-      const yes = window.confirm("Are you sure you want to revoke this hash?");
-      if (yes) {
-        handleCertificateRevoke({
-          storeAddress,
-          fromAddress: adminAddress,
-          certificateHash
-        });
-      }
+  componentDidUpdate(prevProps) {
+    if (prevProps.revokeCertificateHash !== this.props.revokeCertificateHash) {
+      this.setState({
+        certificateHashIsValid: isValidCertificateHash(
```

Comment:
is this costly computation ? if not, why doing this check ?

Reply:
This portion is to check if there are any changes in the redux state for Certificate Hash and do some stuff according.